src/pages/routes.test.ts

```diff
@@ -0,0 +1,44 @@
+import fs from 'fs'
+import parser from 'xml2json'
+
+import { routes } from './RouteDefinitions'
+
+describe('Routes', () => {
+  it('sitemap URLs should exist as Router paths', async () => {
+    const pathNames: string[] = routes.map((routeDef) => routeDef.path)
+    const result = await new Promise<boolean>((resolve) => {
+      fs.readFile('./public/sitemap.xml', 'utf8', (err, data) => {
+        try {
```

Comment:
You can just throw errors to fail the test, instead of `resolve(false)`, and then you can avoid the expectation at the end.